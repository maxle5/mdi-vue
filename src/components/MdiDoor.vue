
<template>
  <svg :width="size" :height="size" :viewBox="viewBox">
      <path d="M8,3C6.89,3 6,3.89 6,5V21H18V5C18,3.89 17.11,3 16,3H8M8,5H16V19H8V5M13,11V13H15V11H13Z" style="fill: currentColor" />
  </svg>
</template>

<script lang="ts">
  import { defineComponent } from 'vue';

  export default defineComponent({
      name: "MdiDoor",
      props: {
          size: { 
              type: Number, 
              default: 24 
          }
      },
      computed: {
        viewBox() {
          return `0 0 ${this.size} ${this.size}`;
        }
      }
  })
</script>
