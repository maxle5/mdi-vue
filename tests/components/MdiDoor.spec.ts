
import { mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import MdiDoor from "../../src/components/MdiDoor.vue";

test("MdiDoor snapshot", () => {
  const wrapper = mount(MdiDoor, {});
  expect(wrapper.html()).toMatchSnapshot();
});
