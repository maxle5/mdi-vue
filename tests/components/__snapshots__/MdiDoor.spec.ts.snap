// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MdiDoor snapshot 1`] = `
"<svg width="24" height="24" viewBox="0 0 24 24">
  <path d="M8,3C6.89,3 6,3.89 6,5V21H18V5C18,3.89 17.11,3 16,3H8M8,5H16V19H8V5M13,11V13H15V11H13Z" style="fill: currentColor;"></path>
</svg>"
`;
